sympy/core/evalf.py

```diff
@@ -710,9 +710,10 @@ def evalf_mul(v: 'Mul', prec: int, options: OPT_DICT) -> TMP_RES:
         man *= m
         exp += e
         bc += b
-        if bc > 3*working_prec:
+        while bc > 3*working_prec:
             man >>= working_prec
             exp += working_prec
+            bc -= working_prec
```

Comment:
I realised while looking at this that `bc` often does not represent the actual bit-count of the mantissa. There must be other places where the arithmetic is off. Maybe in this particular case the issue is that the subs just results in an integer so it can have a bit count that doesn't reflect the actual bit count that would have been generated by evalf for a non-integer expression.

Also it seems that the reason for having `bc` at all is just because it was apparently slow to compute the bit count of an integer but I don't think that's true since Python 3:
https://docs.python.org/3/library/stdtypes.html#int.bit_length

For now it would be good to have some checks in place to see where the bit-count arithmetic is off.

Reply:
> I realised while looking at this that `bc` often does not represent the actual bit-count of the mantissa. There must be other places where the arithmetic is off.

I will look into this soon !

Reply:
Actually it's off right above here. `man *= m` and `bc += b`. The bit count of a product is not always equal to the sum of the bit counts even if zeros are excluded:
```python
In [19]: [
    ...:     ((x.bit_length() + y.bit_length() - (x * y).bit_length()))
    ...:     for x in range(1, 10)
    ...:     for y in range(1, 10)
    ...: ]
Out[19]: 
[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1,
 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 1,
 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
```

Reply:
Probably we should not bother with bit count and just use the integer `bit_length` method to compute them when needed.